### ISSUE TITLE
On/Off toggles are not accessible

### ISSUE THREAD

**Issue by steverep** [litespeedtech/lscache_wp#24]
All of the on/off toggle widgets on the admin pages are not accessible to assistive technology, such as screen readers for the visually impaired.  Please add the appropriate ARIA markup.  Each of these should have a `role="switch"` or `role="checkbox"`, and also use the `aria-checked` attribute.

Please see the following references:
1. [ARIA 1.1 specification of the switch role](https://www.w3.org/TR/wai-aria-1.1/#switch)
2. [ARIA 1.1 Authoring Practice documentation on checkboxes with examples](https://www.w3.org/TR/wai-aria-practices-1.1/#checkbox)

**Comment by hailite** [comment]
Thanks for your suggestion. Please check https://github.com/litespeedtech/lscache_wp/commit/e3f8bb36fbe9a5de234b7721fefae362d68baa27

**Comment by steverep** [comment]
Thanks for the quick response.  A quick scan looks like it might do the trick, but is there a dev URI where I could test it for you?  Or did you test with [NVDA](https://www.nvaccess.org/)?

**Comment by steverep** [comment]
I took a much closer look at this problem after you released these in version 1.8.2.  Lack of ARIA markup is not the issue, so I apologize for steering you in the wrong direction.  In theory, the markup should be fully accessible as radio buttons.  You can safely revert the changes to add `role="switch"` and `aria-checked`.  The native radio input role and `checked` attribute already cover this.

The problem is in the CSS with the following rule:

    .litespeed-switch input {
    	display: none;
    }

Using `display: none` or `visibility: off` effectively hides the actual controls from assistive technology.  Instead, use the `opacity` property to hide the browser's default radio button visual:

    .litespeed-switch input {
    	opacity: 0;
    	cursor: pointer;
    	...

    }

Then you need to incorporate more CSS so that they are positioned properly.  Position them so that they are underneath the respective labels.

**Comment by hailite** [comment]
Thanks for letting us know. Will figure out a way to improve the compatibility, though it's difficult to adjust the styling.
